core/src/main/java/org/apache/carbondata/core/metadata/blocklet/BlockletInfo.java

```diff
@@ -214,6 +218,42 @@ public void setNumberOfPages(int numberOfPages) {
     for (int i = 0; i < mSize; i++) {
       output.writeInt(measureChunksLength.get(i));
     }
+    // Serialize datachunks as well for older versions like V1 and V2
```

Comment:
Can you wrap this logic into a function and mentioning it is for V1 and V2 serialization only, I think it will be more readable

Reply:
ok